From 902dc33f65c3ef7769fa5026769f3c30b75e0f61 Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Mon, 21 Oct 2024 06:45:21 -0700
Subject: [PATCH] umu_run: Rely on an external wrapper to set
 PR_SET_CHILD_SUBREAPER instead of relying on CDLL.

This allows a fully statically linked umu-launcher to still have this prctl call.
---
 umu/umu_run.py | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/umu/umu_run.py b/umu/umu_run.py
index a008230..361343a 100755
--- a/umu/umu_run.py
+++ b/umu/umu_run.py
@@ -650,11 +650,9 @@ def run_in_steammode(proc: Popen) -> int:
 
 def run_command(command: tuple[Path | str, ...]) -> int:
     """Run the executable using Proton within the Steam Runtime."""
-    prctl: CFuncPtr
     cwd: Path | str
     proc: Popen
     ret: int = 0
-    prctl_ret: int = 0
     libc: str = get_libc()
 
     is_gamescope_session: bool = (
@@ -679,18 +677,6 @@ def run_command(command: tuple[Path | str, ...]) -> int:
     else:
         cwd = Path.cwd()
 
-    prctl = CDLL(libc).prctl
-    prctl.restype = c_int
-    prctl.argtypes = [
-        c_int,
-        c_ulong,
-        c_ulong,
-        c_ulong,
-        c_ulong,
-    ]
-    prctl_ret = prctl(PR_SET_CHILD_SUBREAPER, 1, 0, 0, 0, 0)
-    log.debug("prctl exited with status: %s", prctl_ret)
-
     with Popen(
         command,
         start_new_session=True,
-- 
2.47.0

